[EASI-2717] Sanitize CSV input

ClayBenson94 opened the pull request:
# EASI-2717

## Changes and Description

- Added CSV sanitization to intake download

## How to test this change

- Start the app with `scripts/dev up`
- Seed the database with `scripts/dev db:seed`
- Create and submit a new intake that contains `=` and `+` in it
- As an IT Gov admin, go to the home page and download all requests as csv
- Ensure that `+` and `=` characters have slashes before them to ensure that Excel won't parse this data as a formula

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)


## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

adamodd commented:
Haven't looked at the parsing carefully yet (though it looks fine at a glance)

TRB also has some csv downloads. Should that also be patched in this ticket? If so [this line](https://github.com/CMSgov/easi-app/blob/EASI-2717/sanitize_csv/src/views/TechnicalAssistance/TrbAdminTeamHome.tsx#L80) can be
`return cleanCSVData([trbRequestsCsvHeader, ...rows]);`

mynar7 commented:
> Haven't looked at the parsing carefully yet (though it looks fine at a glance)
> 
> TRB also has some csv downloads. Should that also be patched in this ticket? If so [this line](https://github.com/CMSgov/easi-app/blob/EASI-2717/sanitize_csv/src/views/TechnicalAssistance/TrbAdminTeamHome.tsx#L80) can be `return cleanCSVData([trbRequestsCsvHeader, ...rows]);`

Should this be extracted into a utility file then imported and used in both places?

ClayBenson94 commented:
> TRB also has some csv downloads. Should that also be patched in this ticket? If so [this line](https://github.com/CMSgov/easi-app/blob/EASI-2717/sanitize_csv/src/views/TechnicalAssistance/TrbAdminTeamHome.tsx#L80) can be
> `return cleanCSVData([trbRequestsCsvHeader, ...rows]);`

@adamodd I added this in f7cade0 - it's not exactly as you put, as I only clean the row data (and not the headers, though frankly it'd probably work the same)



> Should this be extracted into a utility file then imported and used in both places?

@mynar7 Added this in f7cade0 as well, and added some automated tests to boot 😄

ClayBenson94 commented:
![image](https://user-images.githubusercontent.com/15203744/228847707-6d054a67-9a02-4aa9-8176-54de2d0b920a.png)
 Tested with TRB now too @adamodd ! 🎉

mynar7 reviewed (approved):
LGTM
